### DISCUSSION TITLE
How to add Peripherals view next to Registers view in a custom perspective

### DISCUSSION THREAD

**Discussion by mmx85** [eclipse-embed-cdt/eclipse-plugins#586] [Debug Plug-ins Q&A]
Hi.
I have a custom perspective that previously used GNU MCU Eclipse plug-in 10.0 and my application was able to show Peripherals view and Registers view when I open perspective. After switching to Eclipse Embedded CDT 6.3.2 I have updated viewShortcut in plugin.xml to <viewShortcut id="org.eclipse.embedcdt.internal.debug.gdbjtag.ui.views.PeripheralsView" /> but after that I have Registers view opened only. Peripherals view is missing. Am I doing something wrong or miss something?

Thanks for the answer.

**Comment by TommyMurphyTM1234** [comment]
> Am I doing something wrong or miss something?

You'll probably need to explain in more detail exactly what you're doing and the technical details of your customisations. There's not much info to go on above. You should also check the Eclipse logs in case they contain some useful info

**Reply by mmx85** [reply]
I think I found what was the issue. Before showing "org.eclipse.embedcdt.internal.debug.gdbjtag.ui.views.PeripheralsView" view
there was a check if "ilg.gnumcueclipse.debug.gdbjtag" bundle existed. After changing "ilg.gnumcueclipse.debug.gdbjtag" to "org.eclipse.embedcdt.debug.gdbjtag.core" everything started to work.
Is "org.eclipse.embedcdt.debug.gdbjtag.core" successor of the "ilg.gnumcueclipse.debug.gdbjtag" bundle?

**Reply by ilg-ul** [reply]
I don't know what happened in your case, thus I suggest you install the latest Eclipse, create a new workspace and import your project there. My feeling is that you are using older software.

**Reply by mmx85** [reply]
This is exactly what I'm doing. I'm upgrading my old Eclipse RCP project to the latest plug-ins versions. Since GNU MCU Eclipse is obsoleted I'm switching to Eclipse Embedded CDT and I'm trying to map old ilg.gnumcueclipse IDs and packages to new org.eclipse.embedcdt. But I'm not sure if I'm mapping them correctly since it is not possible just to substitute ilg.gnumcueclipse with org.eclipse.embedcdt.

**Reply by ilg-ul** [reply]
> I'm trying to map old ilg.gnumcueclipse IDs and packages to new org.eclipse.embedcdt

What IDs? Did you try to directly Import the project and it failed?

**Reply by jonahgraham** [reply]
Is it possible @ilg-ul and @mmx85 are talking cross purposes? I think @ilg-ul is talking about user issues where @mmx85 is talking as an extender of Eclipse CDT and Eclipse Embed CDT.

**Reply by jonahgraham** [reply]
> Is "org.eclipse.embedcdt.debug.gdbjtag.core" successor of the "ilg.gnumcueclipse.debug.gdbjtag" bundle?

Yes, sort of. The ilg.gnumcueclipse.debug.gdbjtag was split in https://github.com/eclipse-embed-cdt/eclipse-plugins/commit/93ce442433511ef7237a2e52d1de0f03ab14bd5f into core and ui parts.

So which one you depend on will matter for whether you are doing core or ui stuff.

**Reply by ilg-ul** [reply]
Oops, sorry for the misunderstanding. 

Yes, some plug-ins were split and some IDs were updated to meet the Eclipse Foundation requirements. 

But, as far as I remember, there were no major functional changes, so if you manage to update the dependencies, there are good chances that your plug-in will work.

**Comment by TommyMurphyTM1234** [comment]
> Is it possible @ilg-ul and @mmx85 are talking cross purposes? I think @ilg-ul is talking about user issues where @mmx85 is talking as an extender of Eclipse CDT and Eclipse Embed CDT.

Which is why I asked for a clearer description of the issue in the first place... 🙂
